augur/filter.py

```diff
@@ -1005,6 +1006,16 @@ def get_groups_for_subsampling(strains, metadata, group_by=None):
     if not group_by_set & (set(metadata.columns) | GROUP_BY_GENERATED_COLUMNS):
         raise FilterException(f"The specified group-by categories ({group_by}) were not found.")
 
+    # Warn/error based on other columns grouped with 'week'.
+    if 'week' in group_by_set:
+        if 'year' in group_by_set:
+            print(f"WARNING: 'year' grouping will be ignored since 'week' includes ISO year.", file=sys.stderr)
+            group_by.remove('year')
+            group_by_set.remove('year')
+            generated_columns_requested.remove('year')
+        if 'month' in group_by_set:
+            raise AugurError("'month' and 'week' grouping cannot be used together.")
```

Comment:
It would be nice to warn users about this in the `--group-by` help text.

Reply:
Added in [force-push](https://github.com/nextstrain/augur/compare/f2a6d7182eb86e46ab806f962d7bd30ed3dc18ec..3043718f65410916b498269c85be1b1c0168f013).